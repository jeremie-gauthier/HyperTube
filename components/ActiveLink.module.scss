.active {
  @apply relative font-bold cursor-default;

  .border {
    @apply absolute -bottom-6 w-full bg-white;

    height: 0.33rem;
  }
}

.inactive {
  @apply relative cursor-pointer;

  &:hover {
    .border {
      @apply absolute -bottom-6 w-full bg-white duration-300;

      height: 0.33rem;
    }
  }
}
