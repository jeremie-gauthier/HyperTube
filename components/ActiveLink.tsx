import { useRouter } from "next/router";
import classNames from "classnames";
import Link from "next/link";
import styles from "./ActiveLink.module.scss";

type ActiveLinkProps = {
  children: React.ReactNode;
  href: string;
};

function ActiveLink({ children, href }: ActiveLinkProps): JSX.Element {
  const router = useRouter();
  const isActive = router.pathname === href;
  const activeLink = classNames({
    [styles.inactive]: !isActive,
    [styles.active]: isActive,
  });

  return (
    <Link href={href}>
      <div className={activeLink}>
        {children}
        <div className={styles.border} />
      </div>
    </Link>
  );
}

export default ActiveLink;
