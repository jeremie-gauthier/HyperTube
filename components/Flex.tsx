import React from "react";

type FlexProps = {
  children: React.ReactNode;
} & Partial<DefaultProps>;

type DefaultProps = {
  className: string;
};

export const FlexCol = React.forwardRef<HTMLDivElement, FlexProps>(
  ({ children, className = "", ...rest }, forwardRef) => (
    <div ref={forwardRef} className={`flex flex-col ${className}`} {...rest}>
      {children}
    </div>
  ),
);

export const FlexRow = React.forwardRef<HTMLDivElement, FlexProps>(
  ({ children, className = "", ...rest }, forwardRef) => (
    <div ref={forwardRef} className={`flex flex-row ${className}`} {...rest}>
      {children}
    </div>
  ),
);
