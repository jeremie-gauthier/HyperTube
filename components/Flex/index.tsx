import React from "react";
import styles from "./Flex.module.scss";

type FlexProps = React.HTMLAttributes<HTMLDivElement> & {
  children: React.ReactNode;
} & Partial<DefaultProps>;

type DefaultProps = {
  className: string;
};

export const FlexCol = React.forwardRef<HTMLDivElement, FlexProps>(
  ({ children, className = "", ...rest }, forwardRef) => (
    <div
      ref={forwardRef}
      className={[styles.flexCol, className].join(" ")}
      {...rest}
    >
      {children}
    </div>
  ),
);

export const FlexRow = React.forwardRef<HTMLDivElement, FlexProps>(
  ({ children, className = "", ...rest }, forwardRef) => (
    <div
      ref={forwardRef}
      className={[styles.flexRow, className].join(" ")}
      {...rest}
    >
      {children}
    </div>
  ),
);
