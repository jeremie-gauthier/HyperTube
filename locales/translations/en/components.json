{
	"comment": {
		"confirm_delete": "Confirm the deletion of this comment ?",
		"delete": "Delete comment",
		"delete_success": "Your comment has been removed",
		"wrote": "wrote"
	},
	"navbar": {
		"account": "Account",
		"home": "Home",
		"logout": "Logout",
		"my_list": "My list",
		"search": "Search a movie"
	}
}
