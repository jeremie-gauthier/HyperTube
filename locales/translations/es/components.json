{
	"navbar": {
		"account": "",
		"home": "",
		"logout": "",
		"my_list": "",
		"search": ""
	}
}
