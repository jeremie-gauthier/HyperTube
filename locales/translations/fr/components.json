{
	"comment": {
		"wrote": "a écrit"
	},
	"navbar": {
		"account": "Mon compte",
		"home": "Accueil",
		"logout": "Se déconnecter",
		"my_list": "Ma liste",
		"search": "Rechercher un film"
	}
}
