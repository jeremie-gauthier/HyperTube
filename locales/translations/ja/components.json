{
	"navbar": {
		"account": "アカウント",
		"home": "",
		"logout": "",
		"my_list": "私のリスト",
		"search": "映画を検索する"
	}
}
