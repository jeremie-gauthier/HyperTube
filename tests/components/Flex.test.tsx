import React from "react";
import { FlexCol, FlexRow } from "@/components/Flex";
import renderer from "react-test-renderer";

test("<FlexCol /> should render correctly", () => {
  const tree = renderer
    .create(
      <FlexCol>
        <div />
      </FlexCol>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("<FlexRow /> should render correctly", () => {
  const tree = renderer
    .create(
      <FlexRow>
        <div />
      </FlexRow>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
