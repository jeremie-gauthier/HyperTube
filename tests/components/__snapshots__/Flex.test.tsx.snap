// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexCol /> should render correctly 1`] = `
<div
  className="flex flex-col "
>
  <div />
</div>
`;

exports[`<FlexRow /> should render correctly 1`] = `
<div
  className="flex flex-row "
>
  <div />
</div>
`;
